From 23f6ed6bd31e1f87e80cabe1c585a594ac5ee034 Mon Sep 17 00:00:00 2001
From: Tony Mak <tonymak@google.com>
Date: Thu, 19 Apr 2018 14:24:10 +0100
Subject: [PATCH 186/306] getUserIdLPr should take application id as input

Fixes: 78265824
Test: Setup work profile using TestDPC -> add account into work profile.
Test: Add google.com in personal side to trigger work profile setup.

Change-Id: Ib87301fc98d0ebf1baa043e4badb5144a0f62377
(cherry picked from commit c59dfc61b66b31a2cf221a514599ee672604fdfa)
---
 .../core/java/com/android/server/pm/PackageManagerService.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 50ac4db92cb..9fce12c14bf 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -23603,7 +23603,8 @@ Slog.v(TAG, ":: stepped forward, applying functor at tag " + parser.getName());
 
         private SigningDetails getSigningDetails(int uid) {
             synchronized (mPackages) {
-                final Object obj = mSettings.getUserIdLPr(uid);
+                final int appId = UserHandle.getAppId(uid);
+                final Object obj = mSettings.getUserIdLPr(appId);
                 if (obj != null) {
                     if (obj instanceof SharedUserSetting) {
                         return ((SharedUserSetting) obj).signatures.mSigningDetails;
-- 
2.17.1

